src/Microsoft.AspNetCore.Blazor/Components/ProviderBase.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Blazor.Components
+{
+    /// <summary>
+    /// Non-generic base class for <see cref="Provider{T}"/>.
+    /// </summary>
+    public abstract class ProviderBase
+    {
+        // This base class exists only so that TreeParameterState has a way
+        // to work with all Provider types regardless of their generic param.
+
+        internal abstract bool CanSupplyValue(Type valueType, string providerName);
+    }
```

Comment:
Is the idea of this that the base class is public so that can expose `Provider` but so that no one but us can implement it? Consider an interface as well, which wouldn't need to be public.

Reply:
You're right. I'll do that.

Reply:
Done